src/libraries/System.Threading.RateLimiting/tests/PartitionedRateLimiterTests.cs

```diff
@@ -403,6 +401,36 @@ public async Task Create_CancellationTokenPassedToUnderlyingLimiter()
             await Assert.ThrowsAsync<TaskCanceledException>(async () => await waitTask);
         }
 
+        [Fact]
+        public async Task IdleLimiterIsCleanedUp()
+        {
+            ExpiredLimiter expiredLimiter = null;
+            var factoryCallCount = 0;
+            using var limiter = PartitionedRateLimiter.Create<string, int>(resource =>
+            {
+                return RateLimitPartition.Create(1, _ =>
+                {
+                    factoryCallCount++;
+                    expiredLimiter = new ExpiredLimiter();
+                    return expiredLimiter;
+                });
+            });
+
+            var lease = limiter.Acquire("");
+            Assert.True(lease.IsAcquired);
+
+            Assert.Equal(1, factoryCallCount);
+
+            expiredLimiter.IdleLength = TimeSpan.FromMinutes(1);
+            // Limiter is disposed when timer runs and sees that IdleDuration is greater than idle limit
+            await expiredLimiter.DisposeCts.Task;
+
+            // Acquire will call limiter factory again as the limiter was disposed and removed
+            lease = limiter.Acquire("");
+            Assert.True(lease.IsAcquired);
+            Assert.Equal(2, factoryCallCount);
+        }
+
```

Comment:
Can we add a test verifying that throwing from `DisposeAsync()` doesn't prevent other limiters from being disposed? We should also add a regression test to verify we still dispose limiters if a `TryReplenish()` call throws.

Reply:
> Can we add a test verifying that throwing from DisposeAsync() doesn't prevent other limiters from being disposed?

Could add a test, but it wouldn't reliably test this behavior since we don't control the order of limiters in the dictionary when we flatten it out to a list that we iterate over.

Reply:
Have them all throw and verify they're all still called.